relay-server/src/actors/outcome.rs

```diff
@@ -166,145 +169,147 @@ pub enum DiscardReason {
     ProcessUnreal,
 }
 
+impl Outcome {
+    /// Returns the name of the outcome as recognized by Sentry.
+    fn name(&self) -> &'static str {
+        match self {
+            Outcome::Accepted => "accepted",
+            Outcome::Filtered(_) => "filtered",
+            Outcome::RateLimited(_) => "rate_limited",
+            Outcome::Invalid(_) => "invalid",
+            Outcome::Abuse => "abuse",
+        }
+    }
+
+    fn to_outcome_id(&self) -> u8 {
+        match self {
+            Outcome::Accepted => 0,
+            Outcome::Filtered(_) => 1,
+            Outcome::RateLimited(_) => 2,
+            Outcome::Invalid(_) => 3,
+            Outcome::Abuse => 4,
+        }
+    }
+
+    fn to_reason(&self) -> Option<&str> {
+        match self {
+            Outcome::Accepted => None,
+            Outcome::Invalid(discard_reason) => Some(discard_reason.name()),
+            Outcome::Filtered(filter_key) => Some(filter_key.name()),
+            Outcome::RateLimited(code_opt) => code_opt.as_ref().map(|code| code.as_str()),
+            Outcome::Abuse => None,
+        }
+    }
+}
+
+impl DiscardReason {
```

Comment:
Can you move this under the definition of `DiscardReason` to keep it more local?